exodus_lambda/functions/base.py

```diff
@@ -1,18 +1,37 @@
 import json
+import logging.config
+import logging
 
 
 class LambdaBase(object):
-    def __init__(self, conf_file="lambda_config.json"):
+    def __init__(self, logger_name="default", conf_file="lambda_config.json"):
         self._conf_file = conf_file
         self._conf = None
+        self._logger_name = logger_name
+        self._logger = None
 
     @property
     def conf(self):
         if not self._conf:
             with open(self._conf_file, "r") as json_file:
                 self._conf = json.load(json_file)
-
         return self._conf
 
+    @property
+    def logger(self):
+        if not self._logger:
+            log_conf = self.conf.get("logging", {})
+            logging.config.dictConfig(log_conf)
+            root_logger = logging.getLogger()
+            if not root_logger.handlers:
```

Comment:
This is changing the crash from one kind of exception to another, but in the suggested change I posted, it would avoid the crash.  I'd prefer to do that...

What I'm getting at here is that, even though the lambda runtime is always going to set a handler, making the code crash if that doesn't exist is making the code harder to work with.  For example if you wanted to experiment at the python console or wanted to write a little test script which creates an instance of this class and invokes it with some inputs, then we've made it more difficult to do that right now, as the class now has this hidden dependency on the root logger having a handler.  It only cost a tiny bit of code to avoid that, so I think it's worthwhile.

Reply:
Ah, I understand now. Thanks!